torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2676,6 +2676,24 @@ def _make_tensor_helper(shape, low=None, high=None):
                                 dtypes=[torch.bfloat16])),
                    safe_casts_outputs=True,
                    handles_complex_extremals=False),
+    UnaryUfuncInfo('square',
+                   ref=np.square,
+                   dtypes=all_types_and_complex_and(torch.bool),
+                   dtypesIfCPU=all_types_and_complex_and(torch.bool),
+
+                   dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+                   skips=(
+                       # Reference: https://github.com/pytorch/pytorch/issues/52549
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_hard',
+                                dtypes=[torch.cfloat, torch.cdouble]),
+                       # Reference: https://github.com/pytorch/pytorch/pull/52551#issuecomment-782596181
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_normal',
```

Comment:
Skipping the normal test would be painful because then the forward for this function has to be custom-tested.

Why does the normal test fail? Is it because some of the integer values we insert into the tensor are too large?

Reply:
Actually I have just looked at this PR after a long time. I will be checking the skips in the coming hour or so to check what the issue is and are they even necessary. Will ping you once it is ready :)